src/AudioIOExt.h

```diff
@@ -0,0 +1,60 @@
+/*!********************************************************************
+ 
+ Audacity: A Digital Audio Editor
+ 
+ @file AudioIOExt.h
+ @brief Abstract base class for hooks into audio playback procedures
+ 
+ Paul Licameli
+ 
+ **********************************************************************/
+
+#ifndef __AUDACITY_AUDIO_IO_EXT__
+#define __AUDACITY_AUDIO_IO_EXT__
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include <wx/string.h>
+
+#include "AudioIOBase.h"
+
+struct PaStreamCallbackTimeInfo;
+struct PaStreamInfo;
+struct PlaybackSchedule;
+struct TransportTracks;
+
+class AUDACITY_DLL_API AudioIOExt : public AudioIOExtBase
+{
+public:
+   using Factory = std::function<
+      std::unique_ptr<AudioIOExt>( const PlaybackSchedule& ) >;
+   using Factories = std::vector<AudioIOExt::Factory>;
+   static Factories &GetFactories();
+
+   //! Typically statically constructed
+   struct AUDIO_DEVICES_API RegisteredFactory{
+      explicit RegisteredFactory(Factory factory);
+      ~RegisteredFactory();
+   };
+
+   virtual ~AudioIOExt();
+
+   // Formerly in AudioIoCallback
+   virtual void ComputeOtherTimings(double rate,
```

Comment:
Maybe we should something else in place of *Other*? Like ExtensionStream or whatever?